tests/console/yast2_lan.pm

```diff
@@ -34,18 +34,73 @@ sub handle_dhcp_popup {
     }
 }
 
+
+sub check_etc_hosts_update {
+    my $looprun = 1;
+    assert_script_run "cat /etc/hosts";
+    until ($looprun == 3) {
+        script_run("yast2 lan; echo yast2-lan-status-\$? > /dev/$serialdev", 0);
+        assert_screen "yast2_lan";
+        send_key 'alt-i';    # edit NIC
+        assert_screen 'yast_ncurses_network_card_setup';
+        send_key 'alt-t';    # set to static ip
+        assert_screen 'yast_ncurses_set_static_ip';
+        send_key 'tab';
+        if ($looprun == 1) {
+            send_key_until_needlematch('NICsetup_ncurses_IP_empty', 'backspace');    # "delete existing IP if any
+            type_string "192.168.122.10";
+        }
+        send_key 'tab';
+        # "delete existing netmask if any
+        if ($looprun == 1) {
+            send_key_until_needlematch('NICsetup_ncurses_mask_empty', 'backspace');
+            type_string "/24";
+        }
+        send_key 'tab';
+        send_key_until_needlematch('NICsetup_ncurses_host_empty', 'backspace');
+        type_string "susetest.test$looprun";
+        assert_screen 'yast_ncurses_static_ip_set';
+        send_key 'alt-n';
+        wait_still_screen;
+        send_key 'alt-o';
+        wait_still_screen;
+        wait_serial("yast2-lan-status-0", 180) || die "'yast2 lan' didn't finish";
+        script_run("egrep \"192.168.122.10\\ssusetest.test$looprun\\ssusetest\" /etc/hosts", 30)
+          && die "Expected entry : \"192.168.122.10    susetest.test$looprun susetest\" was not found in /etc/hosts";
+        assert_script_run "cat /etc/hosts";
+        $looprun++;
+    }
+    # Revert changes
```

Comment:
I want the interface to be set back to dhcp at the end of the function because setting a static IP temporarily disables access to network. Regarding the sed command, I will just remove it. Leaving the line in /etc/hosts is either a bug or a feature of yast2 lan, so I tink I should not let it as-is.

Reply:
Right, that's a must have action, but I was referring to [post_run_hook](https://github.com/os-autoinst/openQA/blob/49f37fa5d225c3f111cef0ffef8e0d3154847110/docs/WritingTests.asciidoc#how-to-write-tests). But that's for successful scenarios only. Problem is that if we break network - some tests afterwards can fail too. So, we could use fatal flag or recently introduced `always_rollback` to revert VM state, which works for qemu backends only.